framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -262,17 +291,20 @@ public void updateFromFieldAssignment(
             return;
         }
 
-        String className = getEnclosingClassName(lhs.getElement());
+        ClassSymbol enclosingClass = ((VarSymbol) lhs.getElement()).enclClass();
+        @SuppressWarnings("signature") // https://tinyurl.com/cfissue/3094
+        @DotSeparatedIdentifiers String className = enclosingClass.flatname.toString();
         String jaifPath = storage.getJaifPath(className);
-        AClass clazz = storage.getAClass(className, jaifPath);
+        AClassWrapper clazz = storage.getAClass(className, jaifPath);
+        updateClassMetadata(enclosingClass, clazz);
```

Comment:
I suggest folding this logic into `getAClass`, so that client code here is shorter and (more importantly!) less error-prone.